pkg/ebpf/finding.go

```diff
@@ -50,11 +53,29 @@ func newEvent(id int, name string, s trace.Event) *trace.Event {
 		PodName:             s.PodName,
 		PodNamespace:        s.PodNamespace,
 		PodUID:              s.PodUID,
-		ArgsNum:             s.ArgsNum,
 		ReturnValue:         s.ReturnValue,
 		StackAddresses:      s.StackAddresses,
 		ContextFlags:        s.ContextFlags,
 		MatchedScopes:       s.MatchedScopes,
-		Args:                s.Args,
+		ArgsNum:             len(arguments),
+		Args:                arguments,
 	}
 }
+
+func getArguments(f detect.Finding) []trace.Argument {
+	arguments := make([]trace.Argument, 0, len(f.Data))
+
+	for k, v := range f.Data {
+		arg := trace.Argument{
+			ArgMeta: trace.ArgMeta{
+				Name: k,
+				Type: reflect.TypeOf(v).Name(),
```

Comment:
I think we have specific C types names. Take a look at `GetParamType()` and `ReadArgFromBuff()`.